src/app/test/utils/getIP.ts

```diff
@@ -1,6 +1,189 @@
+import { Agent as HttpAgent } from 'node:http'
+import { Agent as HttpsAgent } from 'node:https'
 import { httpClient } from '../../../utils/httpClient'
 
-const IP_ADDRESS_PROVIDER = 'https://checkip.amazonaws.com'
-export const getIP = async () => {
-  return (await httpClient.get<string>(IP_ADDRESS_PROVIDER).then((res) => res.data)).trim()
+/**
+ * Defines the structure for network metadata configuration, including environment variables
+ * and metadata service providers (IMDS or external providers).
+ */
+interface NetworkMetadataConfig {
+  /**
+   * Predefined public IPv4 address of the server, if available.
+   * If set, no network request will be made to fetch the address.
+   */
+  publicIPv4?: string
+
+  /**
+   * Predefined public IPv6 address of the server, if available.
+   * If set, no network request will be made to fetch the address.
+   */
+  publicIPv6?: string
+
+  /**
+   * Custom IPv4 address provider endpoint (if specified by environment variables).
+   * If set, this will be queried instead of the default metadata provider.
+   */
+  ipv4Provider?: string
+
+  /**
+   * Custom IPv6 address provider endpoint (if specified by environment variables).
+   * If set, this will be queried instead of the default metadata provider.
+   */
+  ipv6Provider?: string
+
+  /**
+   * Default metadata service provider for retrieving the public IPv4 address.
+   * Can be changed to an external service.
+   */
+  defaultIPv4Provider: string
```

Comment:
I don't see the value in this. Why would we need to override `defaultIPvXProvider` values, when we can directly override `ipvXProvider`? I believe we should remove `default...` ones

Reply:
Removed, instead using consts